### DISCUSSION TITLE
Getting the active process name

### DISCUSSION THREAD

**Discussion by Qubik65536** [migueldeicaza/SwiftTerm#304] [Q&A] [answered]
Hi, I'm working on some part of the terminal feature of [CodeEdit Project](https://github.com/CodeEditApp/CodeEdit) in [this PR](https://github.com/CodeEditApp/CodeEdit/pull/1302), and we are trying to make our terminal title updates automatically depending on what's shown in the terminal. What we want exactly is, when a program (e.g. htop, python3, node) is running, our title shows the program name (htop, Python, node, etc.), but when no program is running, the shell name (bash, zsh, etc.) should be displayed.

I am aware of `setTerminalTitle(source:title:)` method of `LocalProcessTerminalViewDelegate` (as https://github.com/migueldeicaza/SwiftTerm/discussions/302 says), but it doesn't always show what we want, and it doesn't work with barebone terminals (for example, zsh without any extension), and what we really want is what macOS's intergrated Terminal.app brings to us as "Active process name", is it possible to get it? For now, we cannot find a way to do so, hope you can steer us in the right direction, much appreciated!

![image](https://github.com/migueldeicaza/SwiftTerm/assets/72877496/a872637e-05ae-4a64-884d-a3f021c8bb2f)
![image](https://github.com/migueldeicaza/SwiftTerm/assets/72877496/f952985e-201f-4780-b331-850182cd65f7)

**Comment by migueldeicaza** [comment]
You need to run a few pre command hooks to do this.   Check the scripts that Apple runs at startup for their terminal (run the command “set” on the command line to quickly see it).

**Reply by Qubik65536** [reply]
> You need to run a few pre command hooks to do this. Check the scripts that Apple runs at startup for their terminal (run the command “set” on the command line to quickly see it).

I do see entries like `precmd_functions`, but I'm not sure what exactly are we looking for, can you give me more advice?

**Reply by austincondiff** [reply]
@migueldeicaza Can you provide any additional clarification here?

**Reply by migueldeicaza** [reply, marked as answer]
My bad, it seems Apple does not do this, but VSCode does:

https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/terminal/browser/media/shellIntegration-bash.sh
